algosdk/encoding.py

```diff
@@ -199,6 +199,25 @@ def encode_address(addr_bytes):
     return _undo_padding(addr.decode())
 
 
+def application_address(application_id: int) -> str:
+    """
+    Calculate an application's account address from its index
+
+    Args:
+        application_index (int): application index on the network
+
+    Returns:
+        str: base32 encoded address for the application's account
+    """
```

Comment:
This is the only new functionality that is truly needed in the Python SDK for C2c. You need to be able to fund an app's account for it to be called by another app. And to do that, you need its address.

Reply:
Not sure where to stick this function. If we had a proper Application class, it would probably go there. Maybe this is a good opportunity to subclass [ABI's ReferenceType](https://github.com/algorand/py-algorand-sdk/blob/6ed0817f67903bb68d516e98f860e79b2839e06f/algosdk/abi/reference.py#L4) with an ApplicationType and have this function live there...